plugins/modules/identity/keycloak/keycloak_user.py

```diff
@@ -0,0 +1,527 @@
+    auth_keycloak_url: https://auth.example.com/auth
+    auth_realm: master
+    auth_username: USERNAME
+    auth_password: PASSWORD
+  delegate_to: localhost
+
+- name: Update the first name of a Keycloak user
+  community.general.keycloak_user:
+    id: '9d59aa76-2755-48c6-b1af-beb70a82c3cd'
+    first_name: an-updated-kc-user-first-name
+    state: present
+    auth_client_id: admin-cli
+    auth_keycloak_url: https://auth.example.com/auth
+    auth_realm: master
+    auth_username: USERNAME
+    auth_password: PASSWORD
+  delegate_to: localhost
+
+- name: Create a keycloak user with credentials
+  community.general.keycloak_user:
+    auth_client_id: admin-cli
+    auth_keycloak_url: https://auth.example.com/auth
+    auth_realm: master
+    auth_username: USERNAME
+    auth_password: PASSWORD
+    username: my-new_user
+    credentials:
+      - priority: 0
+        type: password
+        temporary: false
+        value: 'PASSWORD'
+  delegate_to: localhost
+'''
+
+RETURN = '''
+msg:
+    description: Message as to what action was taken.
+    returned: always
+    type: str
+
+end_state:
+    description: Representation of the user after module execution (sample is truncated).
+    returned: on success
+    type: complex
+    contains:
+        id:
+            description: GUID that identifies the user.
+            type: str
+            returned: always
+            sample: 23f38145-3195-462c-97e7-97041ccea73e
+        username:
+            description: Username of the user.
+            type: str
+            returned: always
+            sample: user-test-123
+        attributes:
+            description: Attributes applied to this user.
+            type: dict
+            returned: always
+            sample:
+                attr1: ["val1", "val2", "val3"]
+
+user:
+  description:
+    - Representation of the user after module execution.
+    - Deprecated return value, it will be removed in community.general 6.0.0. Please use the return value I(end_state) instead.
+  returned: always
+  type: complex
+  contains:
+    id:
+        description: GUID that identifies the user.
+        type: str
+        returned: always
+        sample: 23f38145-3195-462c-97e7-97041ccea73e
+    username:
+        description: Username of the user.
+        type: str
+        returned: always
+        sample: user-test-123
+    attributes:
+        description: Attributes applied to this user.
+        type: dict
+        returned: always
+        sample:
+            attr1: ["val1", "val2", "val3"]
+
+'''
+
+from ansible_collections.community.general.plugins.module_utils.identity.keycloak.keycloak import KeycloakAPI, camel, \
+    keycloak_argument_spec, get_token, KeycloakError
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    """
+    Module execution
+
+    :return:
+    """
+    argument_spec = keycloak_argument_spec()
+
+    meta_args = dict(
+        # Remove alias `username` on `auth_username`
```

Comment:
Could I propose, (not for this PR) we remove the alias, `username` and just require all to use `auth_username` to avoid any ambiguity?

Reply:
Good suggestion, but I think that's a big deal, we can have a dedicated PR for this, how about that?

Reply:
Agree, (hence why I said, "not for this PR").